geospaas_harvesting/ingesters.py

```diff
@@ -121,11 +121,29 @@ def _ingest_dataset(self, url, normalized_attributes):
                 source=source)
 
             #Create the URI for the created Dataset in the database
-            _, created_dataset_uri = DatasetURI.objects.get_or_create(
+            _ , created_dataset_uri = DatasetURI.objects.get_or_create(
                 name=service_name,
                 service=service,
                 uri=url,
                 dataset=dataset)
+
+            for geos_parameter_id in range(len(geos_para_list)):
+                geos_parameter_info = geos_para_list[geos_parameter_id]
+                standard_name = geos_parameter_info.get('standard_name', None)
+                short_name = geos_parameter_info.get('short_name', None)
+                units = geos_parameter_info.get('units', None)
+                if standard_name in ['latitude', 'longitude', None]:
+                    continue
+                params = Parameter.objects.filter(standard_name=standard_name)
+                if params.count() > 1 and short_name is not None:
+                    params = params.filter(short_name=short_name)
+                if params.count() > 1 and units is not None:
+                    params = params.filter(units=units)
+                if params.count() >= 1:
+                    dsp, dsp_created = DatasetParameter.objects.get_or_create(
+                            dataset=dataset, parameter=params[0])
+                    dataset.parameters.add(params[0])
```

Comment:
Isn't it possible to use `update_or_create` instead of all these conditions?

Reply:
@akorosov Please answer us in this regard.

Reply:
Can we use standard name of parameter for searching through those tables of vocabolary?

Can add something to those tables?